src/seedu/addressbook/AddressBook.java

```diff
@@ -558,6 +565,39 @@ private static String getMessageForSuccessfulDelete(String[] deletedPerson) {
         return String.format(MESSAGE_DELETE_PERSON_SUCCESS, getMessageForFormattedPersonData(deletedPerson));
     }
 
+    /**
+     * Deletes a person by the email entered. (assuming that email is unique)
+     *
+     * @param commandArgs full command args string from the user
+     * @return feedback display message for the operation result
+     */
+    private static String executeDeletePersonByEmail(String commandArgs){
+        //retrieves person with email entered
+        final String[] personFound = getPersonWithEmailEntered(commandArgs);
+        if (personFound.length == 0) {
+            return getMessageForInvalidCommandInput(COMMAND_DELETE_BY_EMAIL_WORD, getUsageInfoForDeleteByEmailCommand());
+        }
+        //  returns message + details of person deleted successfully
+        return deletePersonFromAddressBook(personFound) ? getMessageForSuccessfulDelete(personFound) // success
+                : MESSAGE_PERSON_NOT_IN_ADDRESSBOOK; // not found
+    }
+
+    /**
+     * Retrieves all persons in the full model with email entered.
+     *
+     * @param emailEntered of person to delete
+     * @return list of persons in full model with email entered.
+     */
+    private static String[] getPersonWithEmailEntered(String emailEntered) {
+        for (String[] person : getAllPersonsInAddressBook()) {
+            final String email = getEmailFromPerson(person);
+            if (email.equals(emailEntered)) {
+                return person;
```

Comment:
Does this really get all the people with that email? Seems like it just gets the first one.

Reply:
not very sure whether the email is a unique field or not thus I just made my own assumptions because usually emails are unique

Reply:
Ah, then the problem is in the header description, maybe it's clearer to change `Retrieves all persons` -> `Retrieves the person`, I think.